src/components/Text/Text.styles.ts

```diff
@@ -0,0 +1,94 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import { palette, semanticColors } from '../../util/colors';
+import { TextProps, TextSize, TextColor } from './Text';
+import { ellipsisStyle, fontWeightBold } from '../../util/styles';
+import { IRawStyle } from '@uifabric/styling';
+import { CSSProperties } from 'react';
+
+export type Sizes = { [size in TextSize]: string };
+
+export const textColors: { [color in TextColor]: string | undefined } = {
+  primary: semanticColors.bodyText,
+  secondary: palette.neutralPrimaryAlt,
+  metadata: palette.neutralSecondary,
+  white: palette.white,
+  error: semanticColors.errorText,
+  disabled: semanticColors.disabledBodyText,
+};
+
+export const lineHeights: Sizes = {
+  xSmall: '1.6rem',
+  small: '1.6rem',
+  mediumSub: '2.0rem',
+  medium: '2.0rem',
+  large: '2.4rem',
+  xLarge: '2.8rem',
+  xxLarge: '2.8rem',
+};
+
+export const fontSizes: Sizes = {
+  xSmall: '1.0rem',
+  small: '1.2rem',
+  mediumSub: '1.4rem',
+  medium: '1.5rem',
+  large: '1.8rem',
+  xLarge: '2.2rem',
+  xxLarge: '2.4rem',
+};
+
+export const getStyles = (props: TextProps): IRawStyle => {
+  const { size, maxWidth, bold, uppercase, color } = props;
+
+  return {
+    ...(maxWidth ? ellipsisStyle : {}),
+    ...(bold ? fontWeightBold : {}),
```

Comment:
I'd check the compiled output for these spreads. It may be much more costly than the `undefined` approach you're taking below.